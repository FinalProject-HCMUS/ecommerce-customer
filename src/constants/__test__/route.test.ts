import routeConstants from '../route';

describe('routeConstants', () => {
  test('should export an object', () => {
    expect(routeConstants).toBeDefined();
    expect(typeof routeConstants).toBe('object');
  });

  test('should contain top-level keys', () => {
    const expectedKeys = [
      'HOME',
      'SEARCH',
      'AUTH',
      'PRODUCT',
      'PROFILE',
      'CART',
      'CHECKOUT',
      'CHAT',
      'POLICY',
      'VN_PAY_CHECKOUT_RESULT',
      'BLOG',
      'ORDER',
    ];
    expectedKeys.forEach(key => {
      expect(routeConstants).toHaveProperty(key);
    });
  });

  test('AUTH should contain expected auth routes', () => {
    expect(routeConstants.AUTH).toBeDefined();
    expect(routeConstants.AUTH).toHaveProperty('LOGIN');
    expect(routeConstants.AUTH).toHaveProperty('REGISTER');
    expect(routeConstants.AUTH).toHaveProperty('ACTIVATE_ACCOUNT');
    expect(routeConstants.AUTH).toHaveProperty('CONFIRM_EMAIL');
    expect(routeConstants.AUTH).toHaveProperty('CHANGE_PASSWORD');
    expect(routeConstants.AUTH).toHaveProperty('AUTHENTICATE');
    expect(routeConstants.AUTH).toHaveProperty('RESET_PASSWORD_REQUEST');
    expect(routeConstants.AUTH).toHaveProperty('RESET_PASSWORD');
  });

  test('PRODUCT should contain DETAIL', () => {
    expect(routeConstants.PRODUCT).toBeDefined();
    expect(routeConstants.PRODUCT).toHaveProperty('DETAIL');
  });

  test('BLOG should contain LIST and DETAIL', () => {
    expect(routeConstants.BLOG).toBeDefined();
    expect(routeConstants.BLOG).toHaveProperty('LIST');
    expect(routeConstants.BLOG).toHaveProperty('DETAIL');
  });

  test('ORDER should contain ROOT and CONFIRMATION', () => {
    expect(routeConstants.ORDER).toBeDefined();
    expect(routeConstants.ORDER).toHaveProperty('ROOT');
    expect(routeConstants.ORDER).toHaveProperty('CONFIRMATION');
  });

  test('should match snapshot', () => {
    expect(routeConstants).toMatchSnapshot();
  });
});