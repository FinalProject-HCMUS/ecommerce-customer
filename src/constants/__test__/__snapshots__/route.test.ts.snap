// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routeConstants should match snapshot 1`] = `
{
  "AUTH": {
    "ACTIVATE_ACCOUNT": "/activate-account",
    "AUTHENTICATE": "/authenticate",
    "CHANGE_PASSWORD": "/change-password",
    "CONFIRM_EMAIL": "/confirm-email",
    "LOGIN": "/login",
    "REGISTER": "/register",
    "RESET_PASSWORD": "/reset-password",
    "RESET_PASSWORD_REQUEST": "/request-password-reset",
  },
  "BLOG": {
    "DETAIL": "/blog/:id",
    "LIST": "/blog",
  },
  "CART": "/cart",
  "CHAT": "/chat",
  "CHECKOUT": "/checkout",
  "HOME": "/",
  "ORDER": {
    "CONFIRMATION": "/order-confirmation",
    "ROOT": "/orders",
  },
  "POLICY": "/policy",
  "PRODUCT": {
    "DETAIL": "/product/:id",
  },
  "PROFILE": "/profile",
  "SEARCH": "/products",
  "VN_PAY_CHECKOUT_RESULT": "/confirm-vnpay/:statusCode",
}
`;
